runtime/src/attach/bpf_attach_ctx_cuda.cpp

```diff
@@ -134,7 +134,51 @@ void bpf_attach_ctx::start_cuda_watcher_thread()
 							     .map_lookup;
 					auto ptr = bpftime_map_lookup_elem(
 						map_fd, req.key);
-					resp.value = ptr;
+					resp.value = nullptr;
+					if (ptr) {
+						const auto host_ptr =
```

Comment:
What does this piece of code do?

Reply:
> In the trampoline, `_bpf_helper_ext_0001` ultimately returns `resp.map_lookup.value` as a pointer for the GPU code to use (it will subsequently be dereferenced/read/written by the GPU). Previously, `bpftime_map_lookup_elem()` on the syscall-server side returned a pointer within the host process address space. The GPU treated this value as a device pointer, which caused errors like "an illegal memory access was encountered".

I remembered I have mapped the whole segment of boost shared memory to GPU, so it's ok to access bytes in shared memory on GPU. Did you encounter any issues?